src/pages/signin/SignInPage.js

```diff
@@ -64,23 +83,50 @@ const defaultProps = {
 /**
  * @param {Boolean} hasLogin
  * @param {Boolean} hasValidateCode
+ * @param {Object} account
  * @param {Boolean} isPrimaryLogin
- * @param {Boolean} isAccountValidated
+ * @param {Boolean} isUsingMagicCode
+ * @param {Boolean} hasInitiatedSAMLLogin
  * @param {Boolean} hasEmailDeliveryFailure
  * @returns {Object}
  */
-function getRenderOptions({hasLogin, hasValidateCode, hasAccount, isPrimaryLogin, isAccountValidated, hasEmailDeliveryFailure, isClientTheLeader}) {
+function getRenderOptions({hasLogin, hasValidateCode, account, isPrimaryLogin, isUsingMagicCode, hasInitiatedSAMLLogin, isClientTheLeader}) {
+    const hasAccount = !_.isEmpty(account);
+    const isSAMLEnabled = Boolean(account.isSAMLEnabled);
+    const isSAMLRequired = Boolean(account.isSAMLRequired);
+    const hasEmailDeliveryFailure = Boolean(account.hasEmailDeliveryFailure);
+
+    // SAML is temporarily restricted to users on the beta or to users signing in on web and mweb
+    let shouldShowChooseSSOOrMagicCode = false;
+    let shouldInitiateSAMLLogin = false;
+    const platform = getPlatform();
+    if (Permissions.canUseSAML() || platform === CONST.PLATFORM.WEB || platform === CONST.PLATFORM.DESKTOP) {
```

Comment:
Sorry, this went unnoticed as beta is universally true for dev. We need to pass beta props here

Reply:
Discussed 1:1 w/ @MonilBhavsar we're gonna remove the beta code since the ios/android logic is being added [here](https://github.com/Expensify/App/pull/29526)